sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/join/KeyedPCollectionTuple.java

```diff
@@ -154,11 +154,25 @@ public Pipeline getPipeline() {
 
   @Override
   public void finishSpecifying() {
-    for (TaggedKeyedPCollection<K, ?> taggedPCollection : keyedCollections) {
-      taggedPCollection.pCollection.finishSpecifying();
+    // TODO: Make sure key coder is consistent between PCollections. All component PCollections will
```

Comment:
Can you elaborate on this TODO? And same as other comments about why the components will have been finished.